subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/reservation/client/local/LocalReservationClient.kt

```diff
@@ -0,0 +1,157 @@
+package com.avito.instrumentation.reservation.client.local
+
+import com.avito.instrumentation.reservation.adb.AndroidDebugBridge
+import com.avito.instrumentation.reservation.adb.EmulatorsLogsReporter
+import com.avito.instrumentation.reservation.client.ReservationClient
+import com.avito.instrumentation.reservation.request.Device as RequestedDevice
+import com.avito.runner.service.worker.device.Device as WorkerDevice
+import com.avito.instrumentation.reservation.request.Reservation
+import com.avito.instrumentation.util.forEachAsync
+import com.avito.instrumentation.util.iterateInParallel
+import com.avito.instrumentation.util.merge
+import com.avito.runner.service.worker.device.adb.AdbDevicesManager
+import com.avito.utils.logging.CILogger
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.channels.distinctBy
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+class LocalReservationClient(
+    private val androidDebugBridge: AndroidDebugBridge,
+    private val devicesManager: AdbDevicesManager,
+    private val emulatorsLogsReporter: EmulatorsLogsReporter,
+    private val configurationName: String,
+    private val logger: CILogger
+) : ReservationClient {
+
+    private var state: State = State.Idling
+
+    override suspend fun claim(
+        reservations: Collection<Reservation.Data>,
+        serialsChannel: SendChannel<String>,
+        reservationDeployments: SendChannel<String>
```

Comment:
For local deployments it contains stub.
Perhaps, we will store deployment here as a local state.

Reply:
I will try to remove this knowledge from loca reservation later